### PR TITLE
[word lo/hi] support u16 range lookup 

### PR DESCRIPTION
**Pull request by hero78119**
### Description

[_PR description_]

### Issue Link

Close https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1426
Depends by https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1414

### Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Contents

- [x] constraint builder in evm circuit support u16 lookup
- [x] rename `byte_table` to `u8_table` and `query_byte` to `query_u8` so it align with `query_u16` and so on in the future
- [x] refactor state circuit and evm circuit to reuse range table.

### CODE REVIEW

**Comment by ed255** [comment]
This PR touches the state circuit, we will give priority to merging https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1423 first and then update this PR as necessary.

**Comment by ChihChengLiang** [comment]
#1423 Is merged, so this is unblocked.

**Comment by ChihChengLiang** [comment]
@hero78119 Can we resolve conflicts before reviewing it?

**Comment by hero78119** [comment]
> @hero78119 Can we resolve conflicts before reviewing it?

Done! Rebase onto the latest word-lo-hi and resolved conflict

File: zkevm-circuits/src/evm_circuit/execution.rs [hunk 1]
```diff
@@ -829,10 +832,24 @@ impl<F: Field> ExecutionConfig<F> {
...
+                meta.lookup_any("u8 lookup", |meta| {
+                    vec![column.expr()]
+                        .into_iter()
+                        .zip(u8_table.table_exprs(meta).into_iter())
+                        .map(|(expr, table)| (expr, table))
+                        .collect()
+                });
+            }
+        }
+        for column in cell_manager.columns().iter() {
```
Comment (ChihChengLiang) [inline zkevm-circuits/src/evm_circuit/execution.rs:845]: Nitpick: Can we merge this loop with the previous one (in line 834)?

File: zkevm-circuits/src/state_circuit/lookups.rs [hunk 2]
```diff
@@ -86,43 +101,28 @@ impl<F: Field> Chip<F> {
...
+        u8_table.annotate_columns(meta);
+        u10_table.annotate_columns(meta);
+        u16_table.annotate_columns(meta);
         meta.annotate_lookup_any_column(config.call_context_field_tag, || {
             "LOOKUP_call_context_field_tag"
         });
         config
     }
 
     pub fn load(&self, layouter: &mut impl Layouter<F>) -> Result<(), Error> {
```
Comment (ChihChengLiang) [inline zkevm-circuits/src/state_circuit/lookups.rs:125]: Should we have something like `self.config.u8_table.load()` here?

**Review by ChihChengLiang — APPROVED** [review]
LGTM after the feedback is addressed.

File: zkevm-circuits/src/state_circuit/lookups.rs (hunk 2 above)
Comment (hero78119) [reply zkevm-circuits/src/state_circuit/lookups.rs:122]: The different with previous is, ux_table is passed from outside, so the caller will in charge of its table load.
Here the chip load should only responsible for the column by itself.
However I found indeed, table should loaded in state_circuit/dev.rs when it's synthesis as a circuit. Fixed in this commit https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1427/commits/b8e6ad49897de69b2568555b60ffb1224174f906

File: zkevm-circuits/src/evm_circuit/execution.rs (hunk 1 above)
Comment (hero78119) [reply zkevm-circuits/src/evm_circuit/execution.rs]: Sure and good idea! Rewrite this part `match` syntax https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1427/commits/13c03a9343241f2847884990713d4787604388d1

File: zkevm-circuits/src/evm_circuit/util.rs [hunk 3]
```diff
@@ -294,7 +294,9 @@ pub(crate) enum CellType {
     StoragePhase1,
     StoragePhase2,
     StoragePermutation,
-    LookupByte,
+    // TODO combine LookupU8, LookupU16 with Lookup(Table::U8 | Table::U16)
```
[the remark this answers is not in the document]
Comment (hero78119) [reply zkevm-circuits/src/evm_circuit/util.rs:297]: y let create it once this feature sync back to main branch :)

File: zkevm-circuits/src/state_circuit/lookups.rs (hunk 2 above)
Comment (hero78119) [reply zkevm-circuits/src/state_circuit/lookups.rs:122]: Found 3 issue after pr re-trigger ci check (since now we have light test back)
1. `N_U8_LOOKUPS` will complain cell not enough, due to I changed `24 -> 12`, and move half to `N_U16_LOOKUPS`. However we do not cell move to U16 lookup yet. so I will tune it in later PR. For now I set  `N_U16_LOOKUPS=0`, and clippy will complain, therefore adding `#[allow(clippy::reversed_empty_ranges)]` just to silent it.
2. need to adjust test degree for evm circuit, so I hardcode `1<<16` to claim minimum row for evm circuit unittest (otherwise it will failed)

File: zkevm-circuits/src/table/ux_table.rs [hunk 4]
```diff
@@ -0,0 +1,49 @@
+use super::*;
+
+/// Lookup table for max n bits range check
+#[derive(Clone, Copy, Debug)]
+pub struct UXTable<const N_BITS: usize> {
+    col: Column<Fixed>,
+}
+
+impl<const N_BITS: usize> UXTable<N_BITS> {
+    /// Construct the Exponentiation table.
```
Comment (ChihChengLiang) [inline zkevm-circuits/src/table/ux_table.rs:10]: Copy-pasting error?

**Review by ChihChengLiang — APPROVED** [review]
LGTM. Added a nitpick

File: zkevm-circuits/src/table/ux_table.rs (hunk 4 above)
Comment (hero78119) [reply zkevm-circuits/src/table/ux_table.rs]: y and fixed
